examples/drawing/src/undo-redo.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {useEffect, useReducer, useRef} from 'react';
+import {ControlPosition, MapControl, useMap} from '@vis.gl/react-google-maps';
+
+import reducer from './reducer';
+
+import {
+  DrawResult,
+  DrawingActionKind,
+  Overlay,
+  isCircle,
+  isMarker,
+  isPolygon,
+  isPolyline,
+  isRectangle
+} from './types';
+
+interface Props {
+  drawingManager: google.maps.drawing.DrawingManager | null;
+}
+
+export const UndoRedo = ({drawingManager}: Props) => {
+  const map = useMap();
+
+  const [state, dispatch] = useReducer(reducer, {
+    past: [],
+    now: [],
+    future: []
+  });
+
+  const overlaysShouldUpdate = useRef<boolean>(false);
+
+  const pinnedStatesSnapshot = JSON.stringify(
```

Comment:
this would benefit from a comment in why this is here and what the pinnedState is